CMakeLists.txt

```diff
@@ -0,0 +1,145 @@
+cmake_minimum_required(VERSION 3.2)
+project(libxmp VERSION 4.5.0 LANGUAGES C)
+
+set(LIBXMP_DEFINES)
+set(LIBXMP_DEFINES_PUBLIC)
+set(LIBXMP_CFLAGS)
+
+include(build_pros.cmake)
+
+# -fPIC thing
+if(NOT WIN32 AND NOT EMSCRIPTEN AND NOT VITA)
+    option(LIBXMP_PIC   "Build the static library as the position-independent" OFF)
+else()
+    set(LIBXMP_PIC OFF)
+endif()
+# -----------------------------------
+
+if(EMSCRIPTEN OR VITA)
+    set(BUILD_SHARED_DEFAULT OFF)
+else()
+    set(BUILD_SHARED_DEFAULT ON)
+endif()
+
+option(BUILD_STATIC   "Build static library of libXMP" ON)
+option(BUILD_SHARED   "Build shared library of libXMP" ${BUILD_SHARED_DEFAULT})
+if(NOT BUILD_STATIC AND NOT BUILD_SHARED)
+    message(FATAL_ERROR "Both static and shared builds got disabled. You must enable at least one of them.")
+endif()
+
+option(LIBXMP_DISABLE_DEPACKERS     "Disable archive depackers at libXMP" OFF)
+option(LIBXMP_DISABLE_PROWIZARD     "Disable the ProWizard at libXMP" OFF)
+
+include(build_src.cmake)
+
+if(LIBXMP_DISABLE_DEPACKERS)
+    list(APPEND LIBXMP_DEFINES_PUBLIC -DLIBXMP_NO_DEPACKERS)
+else()
+    list(APPEND LIBXMP_SRC_LIST_FULL ${LIBXMP_SRC_LIST_DEPACKERS})
+endif()
+
+if(LIBXMP_DISABLE_PROWIZARD)
+    list(APPEND LIBXMP_DEFINES_PUBLIC -DLIBXMP_NO_PROWIZARD)
+else()
+    list(APPEND LIBXMP_SRC_LIST_FULL ${LIBXMP_SRC_LIST_PROWIZARD})
+endif()
+
+if(MSVC)
+    list(APPEND LIBXMP_DEFINES -Dinline=__inline -D_USE_MATH_DEFINES -DPATH_MAX=260)
+endif()
+
+
+list(APPEND LIBXMP_SRC_FULL ${LIBXMP_SRC_LIST_LITE} ${LIBXMP_SRC_LIST_FULL} ${LIBXMP_SRC_COMMON})
+
+set(XMP_INSTALLS)
+
+# ============= FULL =============
+if(BUILD_STATIC)
+    add_library(xmp_static STATIC ${LIBXMP_SRC_FULL})
+    list(APPEND XMP_INSTALLS xmp_static)
+    set_target_properties(xmp_static PROPERTIES OUTPUT_NAME xmp C_STANDARD 90)
```

Comment:
For symmetry: This directive is missing in lite version

Reply:
Added